data/syntax-highlighting/vim/ftplugin/meson.vim

```diff
@@ -12,9 +12,30 @@ set cpo&vim
 
 setlocal commentstring=#\ %s
 setlocal comments=:#
+setlocal formatoptions+=croql formatoptions-=t
 
-setlocal shiftwidth=2
-setlocal softtabstop=2
+let b:undo_ftplugin = "setl com< cms< fo<"
+
+if get(g:, "meson_recommended_style", 1)
+  setlocal expandtab
+  setlocal shiftwidth=2
+  setlocal softtabstop=2
+  let b:undo_ftplugin .= " | setl et< sts< sw<"
+endif
+
+if exists("loaded_matchit") && !exists("b:match_words")
+  let b:match_words = '\<if\>:\<elif\>:\<else\>:\<endif\>,' .
+	\             '\<foreach\>:\<break\>:\<continue\>:\<endforeach\>'
+  let b:undo_ftplugin .= "| unlet! b:match_words"
+endif
+
+if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
+  let b:browsefilter = "Meson Build Files (meson.build)\tmeson.build\n" .
+	\	       "All Files (*.*)\t*.*\n"
+  let b:undo_ftplugin .= "| unlet! b:browsefilter"
+endif
+
+let b:undo_ftplugin = "setl com< cms< sts< sw<"
```

Comment:
Seems like this is overwriting the previous contents you added.

Reply:
Sure, I think it makes sense to squash it into the commit which obsoleted it as mentioned: "This was retained in error in 695e320"